src/shogun/mathematics/Statistics.h

```diff
@@ -44,6 +45,83 @@ class CStatistics: public CSGObject
 			return sum/vec.vlen;
 		}
 
+	/** Calculates equally weighted cumulative moving average.
+	 * Given \f$cma_{n-1}=\{av_1, ..., av_m\}\f$ and \f$datum=\{x_1, ..., x_m\}\f$,
+	 * this is \f$cma_n=\{av_1+\frac{x_1-av_1}{n}, ..., av_m+\frac{x_m-av_m}{n}\}\f$
+	 *
+	 * @param cma the previous moving average
+	 * @param datum new datum point
+	 * @param n number of data points including the new datum point
+	 * @param result the vector that saves the result 
+	 */
+	template<class T>
+		static void cumulative_moving_average(
+			const SGVector<T>& cma, const SGVector<T>& datum,
+			int32_t n, SGVector<T>& result) 
+		{
+			REQUIRE(
+				n > 0,
+				"Number of data points (%d) must be at least 1",
+				n
+			);
+			T alpha = (T)(1.0) / n;
+			T beta = 1 - alpha;
+			linalg::add<T>(datum, cma, result, alpha, beta);
+		}
```

Comment:
You should do a round of [clang-format](https://clang.llvm.org/docs/ClangFormat.html) at some point to fix the indents!